documentation/modules/security/proc-accessing-kafka-using-ingress.adoc

```diff
@@ -60,30 +77,85 @@ spec:
             host: broker-1.myingress.com
           - broker: 2
             host: broker-2.myingress.com
+          class: nginx  
     # ...
   zookeeper:
     # ...
 ----
-<1> Ingress hosts for the bootstrap service and Kafka brokers.
++
+Add any hostname to the `bootstrap` and `broker-<index>` prefixes that identify the bootstrap and brokers.
++
+NOTE: You might need to add `nginx` as the class if it is missing in the ingresses created. 
 
 . Create or update the resource.
 +
 [source,shell,subs=+quotes]
+----
 kubectl apply -f _<kafka_configuration_file>_
+----
++
+A cluster CA certificate to verify the identity of the kafka brokers is created in the secret `my-cluster-cluster-ca-cert`.
++
+`ClusterIP` type services are created for each Kafka broker, as well as an external bootstrap service.
++
+An `ingress` is also created for each service, with a DNS address to expose them using the NGINX Ingress Controller.
++
+.Ingresses created for the bootstrap and brokers
+[source,shell]
+----
+NAME                        CLASS  HOSTS                    ADDRESS               PORTS
+my-cluster-kafka-0          nginx  broker-0.myingress.com   external.ingress.com  80,443
+my-cluster-kafka-1          nginx  broker-1.myingress.com   external.ingress.com  80,443
+my-cluster-kafka-2          nginx  broker-2.myingress.com   external.ingress.com  80,443
+my-cluster-kafka-bootstrap  nginx  bootstrap.myingress.com  external.ingress.com  80,443
+----
++
+The DNS addresses used for client connection are propagated to the `status` of each ingress.
++
+.Status for the bootstrap ingress
+[source,yaml]
+----
+status:
+  loadBalancer:
+    ingress:
+      - hostname: external.ingress.com
+ # ...
+----
+
+. Use a target broker to check the client-server TLS connection on port 443 using the OpenSSL `s_client`.  
++
+[source,shell]
+----
+openssl s_client -connect my-cluster-kafka-0:443 -servername my-cluster-kafka-0 -showcerts
+----
 +
-`ClusterIP` type services are created for each Kafka broker, as well as an additional _bootstrap service_.
-These services are used by the Ingress controller to route traffic to the Kafka brokers.
-An `Ingress` resource is also created for each service to expose them using the Ingress controller.
-The Ingress hosts are propagated to the `status` of each service.
+The server name is the SNI for passing the connection to the broker. 
 +
-The cluster CA certificate to verify the identity of the kafka brokers is also created in the secret `_<cluster_name>_-cluster-ca-cert`.
+If the connection is successful, the certificates for the broker are returned.
 +
-Use the address for the bootstrap host you specified in the `configuration` and port 443 (_BOOTSTRAP-HOST:443_) in your Kafka client as the _bootstrap address_ to connect to the Kafka cluster.
+.Certificates for the broker
```

Comment:
This worked for routes, but was returning controller certs for ingress. I think it's useful. Assuming it would work in the same way here.

Reply:
Yes, it should work the same way. When it doesn't it typically means that TLS passthrough is not enabled.